### PR TITLE
Feature quick daily flows from graphql

### PR DESCRIPTION
**Pull request by e-kotov**
This PR adds a new way to get daily aggregated data for 2022 onwards with only trip counts via the new GraphQL API endpoint provided by [https://mapas-movilidad.transportes.gob.es/](https://mapas-movilidad.transportes.gob.es/). The default minimum number of trips filter is set only get flows with at least 100 trips to ease the load on the API when users don't change the default.

See the new vignette (included with the PR) on how to use this new function.

### CODE REVIEW

**Comment by e-kotov** [comment]
CMD check is happy, the new function is covered with tests and has as many fail checks on arguments as I could think of. I aim to cover the rest of the user facing functions with similar checks in a separate branch and I think we are ready to submit to CRAN.

File: DESCRIPTION [hunk 1]
```diff
@@ -33,6 +33,7 @@ Imports:
     fs,
     glue,
     here,
+    httr2,
```
Comment (Robinlovelace) [inline DESCRIPTION:36]: 👍 to using this.

File: R/quick-get.R [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+#'   date = "2022-01-01",
+#'   min_trips = 1000
+#' )
+#' }
+#' 
+#' 
+spod_quick_get_od <- function(
+  date = NA,
+  min_trips = 100,
+  distances = c("500m-2km", "2-10km", "10-50km", "50+km"),
```
Comment (Robinlovelace) [inline R/quick-get.R:35]: Great we can select which distances of most relevance.

File: R/quick-get.R [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    variables = list(
+      journeysMunCriteria = journeysMunCriteria
+    )
+  )
+
+  # Send the POST request
+  response <- httr2::request(graphql_endpoint) |>
+    httr2::req_headers(
+      "Content-Type" = "application/json",
+      "User-Agent" = "spanishoddata R package, https://github.com/rOpenSpain/spanishoddata/"
```
Comment (Robinlovelace) [inline R/quick-get.R:200]: Fantastic.

File: _pkgdown.yml [hunk 4]
```diff
@@ -46,6 +47,11 @@ reference:
       - spod_convert
       - spod_connect
       - spod_disconnect
+  - title: "Analysing up to 1 day of trips with no extra variables"
+    desc: >
+      Quickly get a single day of flows between municipalities (without hourly data or any other attributes) for 2022 and onwards
```
Comment (Robinlovelace) [inline _pkgdown.yml:52]: This is so cool.

File: tests/testthat/test-quick-get.R [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  
+  expect_error(
+    spod_quick_get_od(
+      date = "2022-01-01",
+      id_destination = "invalid"
+    ),
+    ".*Invalid municipality IDs detected.*"
+  )
+  
+  expect_error(
```
Comment (Robinlovelace) [inline tests/testthat/test-quick-get.R:85]: Good to see the checks. Excellent work!

File: vignettes/quick-get.qmd [hunk 6]
```diff
@@ -0,0 +1,252 @@
+---
+title: "Quicky get daily data"
```
Comment (Robinlovelace) [inline vignettes/quick-get.qmd:2]: 👍 to the vignette.

File: vignettes/quick-get.qmd [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+    toc: true
+    toc-depth: 2
+    code-overflow: wrap
+execute: 
+  eval: false
+---
+
+# Introduction {#intro}
+
+This vignette demonstrates how to get minimal daily aggregated data on the number of trips between municipalities using the `spod_quick_get_od()` function. With this function, you only get total trips for a single day, and no additional variables that are available in the full [v2 (2022 onwards) data set](v2-2022-onwards-mitma-data-codebook.html). The advantage of this function is that it is much faster than downloading the full data from source CSV files using `spod_get()`, as each CSV file for a single day is about 200 MB in size.
```
Comment (Robinlovelace) [inline vignettes/quick-get.qmd:20]: And uses less memory. But that's kind of self-explanatory. Great intro.

File: vignettes/quick-get.qmd [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+```{r}
+library(spanishoddata)
+library(dplyr)
+```
+
+
+{{< include ../inst/vignette-include/setup-data-directory.qmd >}}
+
+::: callout-note
+Setting a local data directory in this case is optional, as the data is downloaded directly from API and there is no caching on disk. However, the metadata will be downloaded to check the range of valid dates available at the time of the request. So metadata will be downloaded to a temporary location by default, or to the data directory, if you do set it.
```
Comment (Robinlovelace) [inline vignettes/quick-get.qmd:35]:
```suggestion
Setting a local data directory in this case is optional, as the data is downloaded directly from the web API and there is no caching on disk. However, the metadata will be downloaded to check the range of valid dates available at the time of the request. So metadata will be downloaded to a temporary location by default, or to the data directory, if you do set it.
```

**Review by Robinlovelace — APPROVED** [review]
👍

File: vignettes/quick-get.qmd (hunk 7 above)
Comment (e-kotov) [reply vignettes/quick-get.qmd:20]: @Robinlovelace Actually, it may not be obvious to everyone, so I will add this in. Great idea!

**Pull request merged**